invenio_files_rest/ext.py

```diff
@@ -50,6 +52,15 @@ def storage_factory(self):
                 storage_factory
         return self._storage_factory
 
+    @cached_property
+    def upload_factory(self):
```

Comment:
When using ``@cached_property`` the following is enough (note, I've fixed the storage and permission factory in #30). 
```python
imp = self.app.config["FILES_REST_UPLOAD_FACTORY"]
return import_string(imp) if imp else plupload
```